src/main/scala/asciibot/parser/parser.scala

```diff
@@ -0,0 +1,200 @@
+      } else {
+        lines += noComment
+        if (lines.length == 3) {
+          try {
+            rules ++= parseLine(lines)
+          } catch {
+            case CharBlockError(m, sc, ec) => throw new BlockError(m, sc, ec, lineNum-1, lineNum+1)
+          }
+          lines.clear
+        }
+      }
+    } }
+    if (!lines.isEmpty) {
+      throw new FileError("File ended in middle of rule")
+    }
+    rules.toList
+  }
+
+  def parseLine(lns: Seq[String]) : List[Rule] = {
+    assert(lns.length == 3)
+    val maxLen = (lns map (l => l.length)).max
+    // Make sure each string is the same length
+    val nlns = lns map (_.padTo(maxLen, ' '))
+
+    val rules: MutableList[Rule] = new MutableList
+
+    var ruleStart = 0
+
+    // iterate over columns
+    var i = 0
+    for (i <- 0 until maxLen) {
+      // Number of bars in a colum
+      val numBars = (nlns map (l => l(i)) filter (_=='|')).length
+      if (!(numBars == 0 || numBars == 3)) {
+        //throw new RuleSeparatorError(i)
+        throw new CharBlockError("Unaligned vertical bars", i,i+1)
+      }
+      if (numBars == 3) {
+        rules += parseRule(nlns, ruleStart, i)
+        ruleStart = i + 1
+      }
+    }
+    rules += parseRule(nlns, ruleStart, maxLen)
+
+    rules.toList
+  }
+
+  def parseRule(lines: Seq[String], start: Int, end: Int) : Rule = {
+    def throwBE(m : String) = throw new CharBlockError(m, start, end)
+    val arrowStart = lines(1).indexOfSlice("->", start)
+    if (arrowStart < start || arrowStart >= end) {
+      throwBE("No arrow found")
+    }
+    val (inStateStart, inStateEnd) = trimBlock(lines, start, arrowStart)
+    val (outStateStart, outStateEnd) = trimBlock(lines, arrowStart + 2, end) 
+    val (surrs, inState) = parseInState(lines, inStateStart, inStateEnd)
+    val (md, outState) = parseOutState(lines, outStateStart, outStateEnd, surrs)
+    Rule(inState, surrs, md, outState)
+  }
+
+  def trimBlock(lines: Seq[String], start: Int, end: Int) : (Int, Int) = {
+    var s = start
+    var e = end
+    while (lines(0)(s) == ' ' && lines(1)(s) == ' ' && lines(2)(s) == ' ') {
+      s += 1;
+    }
+    while (lines(0)(e - 1) == ' ' && lines(1)(e - 1) == ' ' && lines(1)(e - 1) == ' ') {
+      e -= 1;
+    }
+    (s, e)
+  }
+
+  def parseInState(a: Seq[String], start: Int, end: Int) : (Surroundings, State) = {
+    def throwBE(m : String) = throw new CharBlockError(m, start, end)
+    assert(a.length == 3)
+    val state_str = a(1).slice(start+1,end-1)
+    // Check to make sure there are no illegal characters in the state name
+    if ("#*_|-/> " exists (state_str contains _)) {
+      //throw new StateNameError(state_str)
+      throw new CharBlockError(s"Bad state name `$state_str'", start, end)
+    }
+
+    if (   a(0)(start) != ' '
+        || a(0)(end-1) != ' '
+        || a(2)(start) != ' '
+        || a(2)(end-1) != ' ') {
+        throwBE("Corners of state box not empty")
```

Comment:
There is, we just chose not to use it for a simple operation on 4 items that were not already in a list.